runtime/src/main/java/com/datastax/oss/quarkus/runtime/reactive/QuarkusReactiveCqlSession.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.quarkus.runtime.reactive;
+
+import com.datastax.dse.driver.api.core.cql.reactive.ReactiveRow;
+import com.datastax.dse.driver.api.core.graph.GraphStatement;
+import com.datastax.dse.driver.api.core.graph.reactive.ReactiveGraphNode;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.AsyncResultSet;
+import com.datastax.oss.driver.api.core.cql.PrepareRequest;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.cql.Statement;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+public class QuarkusReactiveCqlSession {
+  private CqlSession cqlSession;
+
+  public QuarkusReactiveCqlSession(CqlSession cqlSession) {
+    this.cqlSession = cqlSession;
+  }
+
+  @NonNull
+  Multi<ReactiveRow> executeContinuouslyReactive(@NonNull String query) {
+    return Wrappers.toMulti(cqlSession.executeContinuouslyReactive(query));
+  }
+
+  @NonNull
+  Multi<ReactiveRow> executeContinuouslyReactive(@NonNull Statement<?> statement) {
+    return Wrappers.toMulti(cqlSession.executeContinuouslyReactive(statement));
+  }
+
+  @NonNull
+  Multi<ReactiveGraphNode> executeReactive(@NonNull GraphStatement<?> statement) {
+    return Wrappers.toMulti(cqlSession.executeReactive(statement));
+  }
+
+  @NonNull
+  Multi<ReactiveRow> executeReactive(@NonNull String query) {
+    return Wrappers.toMulti(cqlSession.executeReactive(query));
+  }
+
+  @NonNull
+  Multi<ReactiveRow> executeReactive(@NonNull Statement<?> statement) {
+    return Wrappers.toMulti(cqlSession.executeReactive(statement));
+  }
+
+  @NonNull
+  Uni<AsyncResultSet> executeAsync(@NonNull String query) {
```

Comment:
I wouldn't expose these methods here, if this class is meant to be fully reactive then we shouldn't expose `AsyncResultSet`.

Reply:
Other extensions (i.e. Mongo) are providing the wrapper around `CompletableFuture` in a Reactive client. If we will not provide it, clients will need to do the transformation on their own so this is a similar situation to `Publisher -> Multi` transformation.
We could introduce yet another bean-like `QuarkusAsyncCqlSession` but I think it is too much.

Reply:
But why would you mix async and reactive in the same application? Here you are just transforming the first `CompletionStage<AsyncResultSet>` into `Uni<AsyncResultSet>`, but as soon as the page arrives, if you need to fetch the next one you will have to call `asyncResultSet.fetchNextPage()` which will give you another `CompletionStage<AsyncResultSet>` back. So in the end, users will have to do the transformation `CompletionStage` -> `Uni` by themselves for every page, except for the first one. Not sure if this is helpful. I would keep this class focused on the bridge between the driver reactive API and Mutiny.